File: frontend/functions/payment.js
exports.handler = async (event, context) => {
  // CORS headers
  const headers = {
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Access-Control-Allow-Methods': 'POST, OPTIONS'
  };

  // Handle preflight requests
  if (event.httpMethod === 'OPTIONS') {
    return {
      statusCode: 200,
      headers,
      body: ''
    };
  }

  if (event.httpMethod !== 'POST') {
    return {
      statusCode: 405,
      headers,
      body: JSON.stringify({ error: 'Method not allowed' })
    };
  }

  try {
    const { plan_id, user_id } = JSON.parse(event.body);

    // Planes de pago
    const payment_plans = [
      {
        id: "monthly",
        name: "Monthly Plan",
        price: 9.99,
        period: "month",
        features: ["Full access to all signals", "Premium features included", "Real-time updates"]
      },
      {
        id: "yearly",
        name: "Yearly Plan",
        price: 99.99,
        period: "year",
        features: ["Full access to all signals", "Premium features included", "Save 17% vs monthly", "Real-time updates"]
      },
      {
        id: "lifetime",
        name: "Lifetime Plan",
        price: 300,
        period: "lifetime",
        features: ["Full access to all signals", "Premium features included", "Best value - One payment forever", "Real-time updates"]
      }
    ];

    const plan = payment_plans.find(p => p.id === plan_id);
    if (!plan) {
      return {
        statusCode: 404,
        headers,
        body: JSON.stringify({ success: false, error: "Plan no encontrado" })
      };
    }

    // Generar enlace de pago
    const payment_id = `pay_${Date.now()}`;

    // URL de Stripe (más confiable que PayPal)
    const stripe_url = `https://buy.stripe.com/test_28o5kK0Xj0Xj0Xj6op?prefilled_email=${user_id}@example.com&client_reference_id=${payment_id}`;

    // URL alternativa de PayPal (formato más simple)
    const paypal_url = `https://www.paypal.me/malukelbasics/${plan.price}`;

    return {
      statusCode: 200,
      headers,
      body: JSON.stringify({
        success: true,
        payment_id: payment_id,
        plan: plan,
        user_id: user_id,
        status: "pending",
        amount: plan.price,
        stripe_url: stripe_url,
        paypal_url: paypal_url,
        message: "Choose your payment method: Stripe (recommended) or PayPal"
      })
    };
  } catch (error) {
    return {
      statusCode: 500,
      headers,
      body: JSON.stringify({
        success: false,
        error: `Error en pago: ${error.message}`
      })
    };
  }
};

File: frontend/api/payment.py
from http.server import BaseHTTPRequestHandler
import json

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        
        # Planes de pago
        plans_data = {
            'success': True,
            'plans': [
                {
                    'id': 'monthly',
                    'name': 'Monthly Plan',
                    'price': '9.99',
                    'period': 'month',
                    'features': [
                        'Full access to all signals',
                        'Real-time technical signals',
                        'Historical signals (last 2 years)',
                        'Fundamental signals (P/E, P/B, ROE)',
                        '45-day calendar forecast',
                        'Email alerts',
                        'Technical support',
                        'Premium features included'
                    ]
                },
                {
                    'id': 'yearly',
                    'name': 'Annual Plan',
                    'price': '99.99',
                    'period': 'year',
                    'features': [
                        'Full access to all signals',
                        'Real-time technical signals',
                        'Historical signals (last 5 years)',
                        'Advanced fundamental signals',
                        '45-day calendar forecast',
                        'Email and SMS alerts',
                        'Priority support',
                        'Custom portfolio analysis',
                        'Save 17% vs monthly'
                    ]
                },
                {
                    'id': 'lifetime',
                    'name': 'Lifetime Plan',
                    'price': '300.00',
                    'period': 'lifetime',
                    'features': [
                        'Full access to all signals',
                        'Real-time technical signals',
                        'Complete historical signals',
                        'Premium fundamental signals',
                        '45-day calendar forecast',
                        'Email, SMS and WhatsApp alerts',
                        'VIP 24/7 support',
                        'Advanced portfolio analysis',
                        'Access to new features',
                        'No renewals needed',
                        'Best value - One payment forever'
                    ]
                }
            ]
        }
        
        self.wfile.write(json.dumps(plans_data).encode())
        return

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        return
